pkg/stdcopy/stdcopy.go

```diff
@@ -155,11 +160,18 @@ func StdCopy(dstout, dsterr io.Writer, src io.Reader) (written int64, err error)
 			}
 		}
 
+		// we might have an error from the daemon mixed up in our multiplexed http
+		// stream. if we do, return it.
+		if out == nil {
```

Comment:
I think it would be a little cleaner to store the result of `StdType(buf[stdWriterFdIndex])` somewhere, and check that against `Systemerr`.